packages/token-details-persister/test/consumerService.integration.test.ts

```diff
@@ -0,0 +1,147 @@
+/* eslint-disable @typescript-eslint/no-floating-promises */
+import { GeneratedTokenAuditDetails, generateId } from "pagopa-interop-models";
+import { afterAll, beforeAll, describe, expect, it, vi } from "vitest";
+import {
+  formatDateyyyyMMdd,
+  formatTimehhmmss,
+  genericLogger,
+} from "pagopa-interop-commons";
+import * as uuidv4 from "uuid";
+import { handleMessages } from ".././src/consumerService.js";
+import { config } from "../src/config/config.js";
+import { fileManager } from "./utils.js";
+
+describe("consumerService", () => {
+  beforeAll(() => {
+    vi.useFakeTimers();
+    vi.setSystemTime(new Date());
+    const uuid = generateId();
+    vi.spyOn(uuidv4, "v4").mockReturnValue(uuid);
+  });
+  afterAll(() => {
+    vi.useRealTimers();
+  });
+
+  it("should write one entry on the bucket", async () => {
+    const auditMessages: GeneratedTokenAuditDetails[] = [getMockAuditDetails()];
+
+    expect(
+      await fileManager.listFiles(config.s3Bucket, genericLogger)
+    ).toMatchObject([]);
+
+    await handleMessages(auditMessages, fileManager, genericLogger);
+
+    const date = new Date();
+    const ymdDate = formatDateyyyyMMdd(date);
+    const hmsTime = formatTimehhmmss(date);
+    const expectedFileName = `${ymdDate}_${hmsTime}_${generateId()}.ndjson`;
+    const expectedFilePathWithFileName = `token-details/${ymdDate}/${expectedFileName}`;
+
+    const fileList = await fileManager.listFiles(
+      config.s3Bucket,
+      genericLogger
+    );
+    expect(fileList).toHaveLength(1);
+    expect(fileList).toMatchObject([expectedFilePathWithFileName]);
+
+    const expectedFileContent = JSON.stringify(auditMessages[0]) + "\n";
+
+    const fileContent = await fileManager.get(
+      config.s3Bucket,
+      expectedFilePathWithFileName,
+      genericLogger
+    );
+
+    const decodedFileContent = new TextDecoder().decode(fileContent);
```

Comment:
very very very minor, but probably a `Buffer.from()` is enough

Reply:
Updated

Reply:
Update: it doesn't work on the CI somehow. Instead locally it works. Reverted the previous version

Reply:
Updated